assets/js/modules/reader-revenue-manager/components/dashboard/RRMSetupSuccessSubtleNotification.js

```diff
@@ -108,13 +111,38 @@ export default function RRMSetupSuccessSubtleNotification( {
 		global.open( serviceURL, '_blank' );
 	};
 
-	const syncPublication = useCallback( () => {
+	const currentOnboardingState = useSelect( ( select ) =>
+		select( MODULES_READER_REVENUE_MANAGER ).getPublicationOnboardingState()
+	);
+
+	const syncPublication = useCallback( async () => {
 		if ( ! shouldSyncPublication ) {
 			return;
 		}
 
-		syncPublicationOnboardingState();
-	}, [ shouldSyncPublication, syncPublicationOnboardingState ] );
+		const { response } = await syncPublicationOnboardingState();
+
+		if ( !! response?.publicationOnboardingState ) {
+			const newOnboardingState = response.publicationOnboardingState;
+
+			if (
+				!! currentOnboardingState &&
+				newOnboardingState !== currentOnboardingState &&
+				newOnboardingState ===
+					PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE
+			) {
+				setValue(
+					UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION,
+					true
+				);
+			}
+		}
```

Comment:
This can be marginally simplified:

```suggestion
		const newOnboardingState = response?.publicationOnboardingState;

		if (
			currentOnboardingState &&
			newOnboardingState !== currentOnboardingState &&
			newOnboardingState ===
				PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE
		) {
			setValue(
				UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION,
				true
			);
		}
```